src/createNavigationContainer.js

```diff
@@ -166,9 +234,9 @@ export default function createNavigationContainer(Component) {
 
     // Per-tick temporary storage for state.nav
 
-    dispatch = action => {
-      if (!this._isStateful()) {
-        return false;
+    dispatch = async action => {
```

Comment:
I don't think we should change this to be async, we depend on it returning `true` or `false` in some situations and users might as well.

Reply:
good point, will fix